tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -38,296 +38,255 @@
 
-    multicopy(inp0, inp1, out=(out0, out1), offset_provider={})
-
-    assert np.allclose(inp0, out0)
-    assert np.allclose(inp1, out1)
+    cases.verify_with_default_data(cartesian_case, multicopy, ref=lambda a, b: (a, b))
 
 
-def test_cartesian_shift(fieldview_backend):
-    a = np_as_located_field(IDim)(np.arange(size + 1, dtype=np.float64))
-    out_I_float = np_as_located_field(IDim)(np.zeros((size,), dtype=float64))
-
+def test_cartesian_shift(cartesian_case):  # noqa: F811 # fixtures
     @field_operator
-    def shift_by_one(inp: Field[[IDim], float64]) -> Field[[IDim], float64]:
+    def shift_by_one(inp: cases.IJKField) -> cases.IJKField:
         return inp(Ioff[1])
 
-    @program(backend=fieldview_backend)
-    def fencil(inp: Field[[IDim], float64], out: Field[[IDim], float64]) -> None:
-        shift_by_one(inp, out=out)
-
-    fencil(a, out_I_float, offset_provider={"Ioff": IDim})
+    inp = cases.allocate(cartesian_case, shift_by_one, "inp").extend({IDim: (0, 1)})()
+    out = cases.allocate(cartesian_case, shift_by_one, cases.RETURN)()
 
-    assert np.allclose(out_I_float.array(), np.arange(1, 11))
+    cases.verify(cartesian_case, shift_by_one, inp, out=out, ref=inp[1:])
 
 
-def test_unstructured_shift(reduction_setup, fieldview_backend):
-    Vertex = reduction_setup.Vertex
-    Edge = reduction_setup.Edge
-    E2V = reduction_setup.E2V
-
-    a = np_as_located_field(Vertex)(np.zeros(reduction_setup.num_vertices))
-    b = np_as_located_field(Edge)(np.zeros(reduction_setup.num_edges))
-
-    @field_operator(backend=fieldview_backend)
-    def shift_by_one(inp: Field[[Vertex], float64]) -> Field[[Edge], float64]:
+def test_unstructured_shift(unstructured_case):  # noqa: F811 # fixtures
+    @field_operator
+    def shift_by_one(inp: cases.VField) -> cases.EField:
         return inp(E2V[0])
 
-    shift_by_one(a, out=b, offset_provider={"E2V": reduction_setup.offset_provider["E2V"]})
+    inp = cases.allocate(unstructured_case, shift_by_one, "inp")()
+    out = cases.allocate(unstructured_case, shift_by_one, cases.RETURN)()
 
-    ref = np.asarray(a)[reduction_setup.offset_provider["E2V"].table[slice(0, None), 0]]
+    ref = np.asarray(inp)[unstructured_case.offset_provider["E2V"].table[slice(0, None), 0]]
+    cases.verify(unstructured_case, shift_by_one, inp, out=out, ref=ref)
 
-    assert np.allclose(b, ref)
 
-
-def test_fold_shifts(fieldview_backend):
+def test_fold_shifts(cartesian_case):  # noqa: F811 # fixtures
     """Shifting the result of an addition should work."""
-    a = np_as_located_field(IDim)(np.arange(size + 1, dtype=np.float64))
-    b = np_as_located_field(IDim)(np.ones((size + 2)) * 2)
-    out_I_float = np_as_located_field(IDim)(np.zeros((size,), dtype=float64))
 
     @field_operator
-    def auto_lift(
-        inp1: Field[[IDim], float64], inp2: Field[[IDim], float64]
-    ) -> Field[[IDim], float64]:
+    def auto_lift(inp1: cases.IJKField, inp2: cases.IJKField) -> cases.IJKField:
```

Comment:
If test is already described by the name of the test function, I'd use a generic name in the field_operator, it seems we are often copy-pasting anyway.
```suggestion
    def testee(inp1: cases.IJKField, inp2: cases.IJKField) -> cases.IJKField:
```

Reply:
Agree